plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java

```diff
@@ -0,0 +1,295 @@
+        Setting.Property.NodeScope);
+
+    static final Setting<Integer> EVENT_LOOP_THREAD_COUNT = Setting.intSetting(
+        "repository.azure.http_client.event_loop_executor_thread_count",
+        DEFAULT_EVENT_LOOP_THREAD_COUNT,
+        Setting.Property.NodeScope);
+
+    static final Setting<Integer> MAX_OPEN_CONNECTIONS = Setting.intSetting(
+        "repository.azure.http_client.max_open_connections",
+        DEFAULT_MAX_CONNECTIONS,
+        Setting.Property.NodeScope);
+
+    static final Setting<TimeValue> OPEN_CONNECTION_TIMEOUT = Setting.timeSetting(
+        "repository.azure.http_client.connection_timeout",
+        DEFAULT_CONNECTION_TIMEOUT,
+        Setting.Property.NodeScope);
+
+    static final Setting<TimeValue> MAX_IDLE_TIME = Setting.timeSetting(
+        "repository.azure.http_client.connection_max_idle_time",
+        DEFAULT_MAX_CONNECTION_IDLE_TIME,
+        Setting.Property.NodeScope);
+
+    static final Setting<String> REACTOR_SCHEDULER_EXECUTOR_NAME = Setting.simpleString(
+        "repository.azure.http_client.reactor_executor_name",
+        REPOSITORY_THREAD_POOL_NAME,
+        Setting.Property.NodeScope);
+
+    private final ThreadPool threadPool;
+    private final String reactorExecutorName;
+    private final EventLoopGroup eventLoopGroup;
+    private final ConnectionProvider connectionProvider;
+    private final ByteBufAllocator byteBufAllocator;
+    private final ClientLogger clientLogger = new ClientLogger(AzureClientProvider.class);
+    private volatile boolean closed = false;
+
+    AzureClientProvider(ThreadPool threadPool,
+                        String reactorExecutorName,
+                        EventLoopGroup eventLoopGroup,
+                        ConnectionProvider connectionProvider,
+                        ByteBufAllocator byteBufAllocator) {
+        this.threadPool = threadPool;
+        this.reactorExecutorName = reactorExecutorName;
+        this.eventLoopGroup = eventLoopGroup;
+        this.connectionProvider = connectionProvider;
+        this.byteBufAllocator = byteBufAllocator;
+    }
+
+    static int eventLoopThreadsFromSettings(Settings settings) {
+        return EVENT_LOOP_THREAD_COUNT.get(settings);
+    }
+
+    static AzureClientProvider create(ThreadPool threadPool, Settings settings) {
+        final ExecutorService executorService;
+        try {
+            executorService = threadPool.executor(EVENT_LOOP_EXECUTOR.get(settings));
+        } catch (IllegalArgumentException e) {
+            throw new SettingsException("Unable to find executor [" + EVENT_LOOP_EXECUTOR.get(settings) + "]");
+        }
+        // Most of the code that needs special permissions (i.e. jackson serializers generation) is executed
+        // in the event loop executor. That's the reason why we should provide an executor that allows the
+        // execution of privileged code
+        final EventLoopGroup eventLoopGroup = new NioEventLoopGroup(eventLoopThreadsFromSettings(settings),
```

Comment:
As I point out here most of the response handling code is executed in the netty event loop executor, meaning that we should provide a privileged environment to the tasks executed in that event loop. 
I tried different approaches to solve this issue, one of them was to create a wrapper around 
 `com.azure.core.http.HttpClient` that would publish the emitted response in a different thread pool:

```java
Mono<HttpResponse> send(HttpRequest request) {
         return delegate.send(request).publishOn(privilegedExecutor);
}
```

But it seems like the `HttpResponse` is parsed before, so that didn't help. 

Maybe I'm missing something here and we can provide a more fine-grained solution.